lighthouse-core/runner.js

```diff
@@ -19,169 +18,168 @@ const path = require('path');
       const resultsById = {};
-      for (const audit of runResults.auditResults) resultsById[audit.name] = audit;
+      for (const audit of auditResults) resultsById[audit.name] = audit;
 
-      let categories;
+      let reportCategories;
       if (opts.config.categories) {
-        Runner._scoreAndCategorize(opts, resultsById);
-        categories = Object.values(opts.config.categories);
+        reportCategories = ReportScoring.scoreAllCategories(opts.config.categories, resultsById);
       }
       return {
-        userAgent: runResults.artifacts.UserAgent,
-        lighthouseVersion: require('../package').version,
-        fetchedAt: runResults.artifacts.fetchedAt,
+        userAgent: artifacts.UserAgent,
+        lighthouseVersion,
+        fetchedAt: artifacts.fetchedAt,
         generatedTime: 'Please use .fetchedAt instead',
         initialUrl: opts.initialUrl,
         url: opts.url,
         runWarnings: lighthouseRunWarnings,
         audits: resultsById,
-        artifacts: runResults.artifacts,
-        runtimeConfig: Runner.getRuntimeConfig(opts.settings),
-        reportCategories: categories,
+        artifacts,
+        runtimeConfig: Runner.getRuntimeConfig(settings),
+        reportCategories,
         reportGroups: opts.config.groups,
       };
-    }).catch(err => {
-      return Sentry.captureException(err, {level: 'fatal'}).then(() => {
-        throw err;
-      });
-    });
-
-    return run;
-  }
-
-  /**
-   * No browser required, just load the artifacts from disk
-   * @param {string} path
-   * @return {!Promise<!Artifacts>}
-   */
-  static _loadArtifactsFromDisk(path) {
-    return assetSaver.loadArtifacts(path);
+    } catch (err) {
+      // @ts-ignore TODO(bckenny): Sentry type checking
+      await Sentry.captureException(err, {level: 'fatal'});
+      throw err;
+    }
   }
 
   /**
    * Establish connection, load page and collect all required artifacts
-   * @param {*} opts
-   * @param {*} connection
-   * @return {Promise<Artifacts>}
+   * @param {{config: LH.Config, url: string, initialUrl?: string, driverMock?: Driver}} runnerOpts
+   * @param {Connection} connection
+   * @return {Promise<LH.Artifacts>}
    */
-  static async _gatherArtifactsFromBrowser(opts, connection) {
-    if (!opts.config.passes) {
+  static async _gatherArtifactsFromBrowser(runnerOpts, connection) {
+    if (!runnerOpts.config.passes) {
       throw new Error('No browser artifacts are either provided or requested.');
     }
 
-    opts.driver = opts.driverMock || new Driver(connection);
-    return GatherRunner.run(opts.config.passes, opts);
-  }
-
-  /**
-   * Save collected artifacts to disk
-   * @param {!Artifacts} artifacts
-   * @param {string} path
-   * @return {!Promise>}
-   */
-  static _saveArtifacts(artifacts, path) {
-    return assetSaver.saveArtifacts(artifacts, path);
+    const driver = runnerOpts.driverMock || new Driver(connection);
+    const options = {
```

Comment:
I see you already renamed `opts` to `runnerOpts`, though in the parent context is still has the `opts` name. 

Regardless, feels like we can do `gatherOpts` for this guy. eviscerate all the `options`. woo!

Reply:
boom done